src/dmd/typesem.d

```diff
@@ -3223,7 +3223,16 @@ private extern(C++) final class DotExpVisitor : Visitor
                 }
                 checkAccess(e.loc, sc, null, v);
                 Expression ve = new VarExp(e.loc, v);
-                ve = ve.expressionSemantic(sc);
+                if (e.op == TOK.type || e.op == TOK.this_ || e.op == TOK.super_ || e.op == TOK.variable)
+                {
+                    ve = ve.expressionSemantic(sc);
```

Comment:
The point here is to avoid being forced to optimize the expression if we know the lhs of the comma expression would be trivial, since optimization may change the meaning but is sometimes necessary so we don't have a pointless comma expression lying around. I wish there was a better way to do this.

Reply:
Please add this explanation as a comment in the code. It will be useful in the future when people will try to understand the code.

Reply:
Documentation added.